### DISCUSSION TITLE
Notify systemd via sd_notify

### DISCUSSION THREAD

**Discussion by xanderio** [seaweedfs/seaweedfs#3439] [Ideas]
## Problem
I've noticed that e.g. the master server take quite a bit of time (22s on my machine) until it response to any request. 

I would like to use the systemd unit dependency feature to start the volume and filer service only after the master server is ready to process requests.

Currently this is not really possible to do, as systemd thinks the unit is up and running directly after the process is started. 

## Solution 
To fix this seaweedfs should notify systemd of its status via the notify socket.

documentation: https://www.man7.org/linux/man-pages/man3/sd_notify.3.html

**Comment by kmlebedev** [comment]
Maybe systemd can pull status from http healze URL?

**Reply by xanderio** [reply]
That is not something that systemd is capable of doing.

**Comment by LuckyFrost** [comment]
Type=forking more common and simpler to do.
